src/main/java/com/xemantic/ankh/shared/presenter/Presenter.java

```diff
@@ -38,25 +39,42 @@
  */
 public abstract class Presenter {
 
-  private final List<Disposable> subscriptions = new LinkedList<>();
+  private final List<Observable<?>> observables = new LinkedList<>();
 
-  protected <T> CallDefiner<T> on(Observable<T> observable) {
-    Objects.requireNonNull(observable);
-    return consumer ->
-        subscriptions.add(
+  private Action onStart = Functions.EMPTY_ACTION;
+
+  private List<Disposable> subscriptions;
+
+  protected Presenter(Observable<?> ... observables) {
+    this.observables.addAll(Arrays.asList(observables));
+  }
+
+  protected void register(Observable<?> observable) {
+    observables.add(observable);
+  }
+
+  protected void onStart(Action onStart) {
+    this.onStart = onStart;
+  }
+
+  public void start() {
+    subscriptions = Observable.fromIterable(observables)
+        .map(observable ->
             observable
                 .retry(throwable -> { // always retry, will cause unconditional resubscription
                   Errors.onError(throwable);
                   return true;
                 })
-                .subscribe(value -> {
-                  try {
-                    consumer.accept(value);
-                  } catch (Exception e) { // prevent unsubscription but handle exception
-                    Errors.onError(e);
-                  }
-                })
-    );
+                .subscribe()
+        )
+        .toList()
+        .blockingGet();
```

Comment:
this might work, but it is extremely dangerous, even in the JVM blocking is discouraged, so in JS is even worst. I make it compile in GWT just bc it might be helpful to make tests, but I never thought it will be used in actual code. You can just use java stream, ix (interactiveX, like reactivex but for iterables instead of callbacks https://github.com/akarnokd/ixjava), or even easier, just a for each loop (either with expression or with lambda) `attachObservables.forEach((o) -> subscribe(o, attachSubscriptions, "attach"));`.

Reply:
I might have problems with Java 8 API support on android. At least lambdas should work. I already changed every `Iterable.forEach` to standard for loop. I also changed this code.